waiter/test/waiter/scheduler/kubernetes_test.clj

```diff
@@ -149,6 +152,30 @@
         (clojure.data/diff expected# actual#)))
      (is (= expected# actual#))))
 
+(deftest test-replicaset-spec-pre-stop-cmd
```

Comment:
I've only added unit tests here. It's unclear how to integration test this behavior. I've tried writing a test for running a long-running request, but it's not clear how we can force scale down for a specific instance. Right now we only have an inter-router endpoint for killing any killable instance of the service-id